subprojects/core/src/integTest/groovy/org/gradle/api/HttpScriptPluginIntegrationSpec.groovy

```diff
@@ -84,8 +86,9 @@ class HttpScriptPluginIntegrationSpec extends AbstractIntegrationSpec {
 
         then:
         executer.expectDeprecationWarning("Applying script plugins from insecure URIs has been deprecated. This is scheduled to be removed in Gradle 7.0. " +
-                "The provided URI '${server.uri("/external.gradle")}' uses an insecure protocol (HTTP). " +
-                "Use '${GUtil.toSecureUrl(server.uri("/external.gradle"))}' instead or try 'apply from: resources.text.fromInsecureUri(\"${server.uri("/external.gradle")}\")' to silence the warning.")
+            "The provided URI '${server.uri("/external.gradle")}' uses an insecure protocol (HTTP). " +
+            "Use '${GUtil.toSecureUrl(server.uri("/external.gradle"))}' instead or try 'apply from: resources.text.fromInsecureUri(\"${server.uri("/external.gradle")}\")' to silence the warning. " +
+            "See https://docs.gradle.org/${GradleVersion.current().version}/dsl/org.gradle.api.resources.TextResourceFactory.html#org.gradle.api.resources.TextResourceFactory:fromInsecureUri(java.lang.Object) for more details.")
```

Comment:
I don't have a much better idea than what you're already doing, but would a helper method to build this URL help?

e.g.,
```
"See ${docs("dsl/org.gradle.api.resources.TextResourceFactory.html#org.gradle.api.resources.TextResourceFactory:fromInsecureUri(java.lang.Object)")} for more details."
```
I don't like the complication here to get the quotes, parentheses and curly braces to line up.

Would it be helpful to massage the text that's seen from the build in `expectDeprecationWarning` so that we can use a fixed Gradle version always:
```
"See https://docs.gradle.org/current/dsl/org.gradle.api.resources.TextResourceFactory.html#org.gradle.api.resources.TextResourceFactory:fromInsecureUri(java.lang.Object)")} for more details."
```

I think it's a little too implementation specific to need to build this URL with the Gradle version, so if we could have a static string I think it would make it easier to see what we're linking to.

Reply:
Let's try this maybe? https://github.com/gradle/gradle/pull/11931/commits/e1eb1dc2be8d92486eaa0ca627d01a1d4e91e561